### PR TITLE
Block Editor: Remove the Shuffle block toolbar button

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
See #63900.

PR removes the Shuffle block toolbar button from the block editors' default viewing mode.

## Why?
It's a destructive action that shouldn't be this easy to get to.

## Testing Instructions
1. Open a post or page.
2. Insert a pattern from a category - "Call to Action"
3. Confirm that the "Shuffle" button isn't still available in the toolbar.
4. Switch to Zoom out view - "Desktop (50%)".
5. Confirm that the "Shuffle" button is visible.

### Testing Instructions for Keyboard
Same.

## Screenshots or screencast 
CleanShot 2024-08-31 at 09 57 51

### CODE REVIEW

File: packages/block-editor/src/components/block-toolbar/index.js [hunk 1]
```diff
@@ -106,7 +101,6 @@ export function PrivateBlockToolbar( {
 			isDefaultEditingMode: _isDefaultEditingMode,
 			blockType: selectedBlockClientId && getBlockType( _blockName ),
 			shouldShowVisualToolbar: isValid && isVisual,
-			rootClientId: blockRootClientId,
```
Comment (Mamaduka) [inline packages/block-editor/src/components/block-toolbar/index.js:109]: Unrelated change: This value hasn't been used since #19596.

**Review by cbravobernal — APPROVED** [review]
Works as expected.

**Comment by annezazu** [comment]
Connecting this PR to this issue for additional feedback and context: https://github.com/WordPress/gutenberg/issues/62288

**Comment by fabiankaegy** [comment]
Hey @Mamaduka   👋 

Would you be able to help write a dev note for this for the 6.7 release? We are planning to have this as part of a larger Miscellaneous Editor Updates note.

We are hoping to get all drafts in by October 13th to leave some time for reviews before the RC1. 

All Dev Notes get tracked in #65784 so feel free to leave a note there or ping me directly :) 

Please let us know if you can assist with that.

Thanks in advance :)

**Comment by Mamaduka** [comment]
I don't know if this needs a dev note or a mention in one. We add/remove parts of UI controls all the time and don't publish dev notes.

**Comment by fabiankaegy** [comment]
> I don't know if this needs a dev note or a mention in one. We add/remove parts of UI controls all the time and don't publish dev notes.

@Mamaduka I agree we don't do it very often for UI changes. But broadly speaking we should do it more often. 

And especially in this case where the shuffle button was quite prominently featured in the toolbar I'd expect this to be a feature every user has at least encountered a few times. So I personally do think a dev note (as a small note in the overall miscellaneous section ) would be nice :) 

But also happy to drop if if you feel strongly about it :)

**Comment by Mamaduka** [comment]
I always understood that dev notes are for changes that affect developers/consumers. I don't see how this UI change might affect them. It was never actionable for developers.

**Comment by fabiankaegy** [comment]
Sounds good  👍 removing it from the list then.
